pkg/util/config/feature_config.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+const (
+	// known keys
+	enableAllAlphaFeatures = "enableAllAlphaFeatures"
+)
+
+var featureDefaults = map[string]bool{
+	enableAllAlphaFeatures: false,
+}
+
+type FeatureConfig interface {
+	ConfigurationMap() ConfigurationMap
+	AddFlag(fs *pflag.FlagSet)
+	GetEnableAllAlphaFeatures() (bool, error)
+	// TODO: Define accessors for each non-API alpha feature.
+}
+
+type featureConfig struct {
+	configMap *ConfigurationMap
+}
+
+func NewFeatureConfig() FeatureConfig {
+	m := make(ConfigurationMap)
+	return &featureConfig{
+		configMap: &m,
+	}
+}
+
+func (f *featureConfig) Set(value string) error {
```

Comment:
does this "set the flag"?  Why is it a string?  Is it because it is implementing the flag interface?  Comments...